compiler/lib/parse_bytecode.ml

```diff
@@ -2088,64 +2112,82 @@ and compile infos pc state instrs =
           :: Let (meths, Field (obj, 0))
           :: instrs)
     | STOP -> instrs, Stop, state
-    | RESUME ->
-        let _stack = State.accu state in
-        let _func = State.peek 0 state in
-        let _arg = State.peek 1 state in
-        let state = State.pop 2 state in
+    | (RESUME | RESUMETERM | PERFORM | REPERFORMTERM) when not (Config.Flag.effects ()) ->
+        if not !effect_warning
+        then (
+          warn
```

Comment:
Should we move this warning in generate, after we give DCE a chance to remove usage of effects ?

Reply:
We can do that, indeed.

Reply:
I've pushed this change